vm/src/pyobject.rs

```diff
@@ -1339,7 +1339,30 @@ pub trait PyValue: fmt::Debug + PyThreadingConstraint + Sized + 'static {
 
 pub trait BorrowValue<'a>: PyValue {
     type Borrowed: 'a + Deref;
+
     fn borrow_value(&'a self) -> Self::Borrowed;
+
+    fn copied_value(&'a self) -> <Self::Borrowed as Deref>::Target
+    where
+        <Self::Borrowed as Deref>::Target: Copy,
+    {
+        *self.borrow_value()
+    }
+
+    fn cloned_value(&'a self) -> <Self::Borrowed as Deref>::Target
+    where
+        <Self::Borrowed as Deref>::Target: Clone,
+    {
+        self.borrow_value().clone()
+    }
+
+    fn owned_value<Owned>(&'a self) -> Owned
+    where
+        Owned: std::borrow::Borrow<<Self::Borrowed as Deref>::Target>,
+        <Self::Borrowed as Deref>::Target: ToOwned<Owned = Owned>,
```

Comment:
I think you don't need a generic for this, can you do `-> <<Self::Borrowed as Deref>::Target as ToOwned>::Owned where <Self:: Borrowed as Deref>::Target: ToOwned` ?